RIOT-OS/RIOT
# cpu/esp32: fix esp_wifi_enterprise compilation problem

### Contribution description

This PR fixes the compilation problem for module `esp_wifi_enterprise` reported in issue #17305.

With the change from using archive files to using object files in PR #14754, the module `esp_wifi_enterprise` was no longer compilable. The reason for this was that the file `bignum.c` was present in two different vendor sub-directories and created two identical object files, which then led to a symbol conflict when linking. This PR removes one of these identical files. The one that is used in all `esp_wifi` variants is kept, the one that is only used in `esp_wifi_enterprise` is dropped.

### Testing procedure

Compile
```
CFLAGS='-DESP_WIFI_EAP_USER=\"riot\" -DESP_WIFI_EAP_PASS=\"riot\"' USEMODULE=esp_wifi_enterprise make BOARD=esp32-wroom-32 -C examples/gnrc_networking
```
Without this error it should show multiple definition errors like
```
examples/gnrc_networking/bin/esp32-wroom-32/esp_idf_wpa_supplicant_wpa2_tls/bignum.o: In function `bignum_init':
cpu/esp32/vendor/esp-idf/wpa_supplicant/src/wpa2/tls/bignum.c:40: multiple definition of `bignum_init'
examples/gnrc_networking/bin/esp32-wroom-32/esp_idf_wpa_supplicant_crypto/bignum.ocpu/esp32/vendor/esp-idf/wpa_supplicant/src/crypto/bignum.c:40: first defined here
```
With this PR the compilation should be successful. Since there is no code change successful compilation should be enough.
 
### Issues/PRs references

Fixes #17305
